src/components/Popover/Popover.tsx

```diff
@@ -78,128 +80,149 @@ export interface PopoverProps {
 }
 
+export interface PopoverPublicAPI {
+  forceUpdatePosition(): void;
+}
+
 // TypeScript can't generate types that correctly infer the typing of
 // subcomponents so explicitly state the subcomponents in the type definition.
 // Letting this be implicit works in this project but fails in projects that use
 // generated *.d.ts files.
 
-export const Popover: React.FunctionComponent<PopoverProps> & {
-  Pane: typeof Pane;
-  Section: typeof Section;
-} = function Popover({
-  activatorWrapper = 'div',
-  children,
-  onClose,
-  activator,
-  preventFocusOnClose,
-  active,
-  fixed,
-  ariaHaspopup,
-  preferInputActivator = true,
-  colorScheme,
-  zIndexOverride,
-  ...rest
-}: PopoverProps) {
-  const [activatorNode, setActivatorNode] = useState<HTMLElement>();
-  const activatorContainer = useRef<HTMLElement>(null);
-  const WrapperComponent: any = activatorWrapper;
-  const id = useUniqueId('popover');
-
-  const setAccessibilityAttributes = useCallback(() => {
-    if (activatorContainer.current == null) {
-      return;
-    }
-
-    const firstFocusable = findFirstFocusableNodeIncludingDisabled(
-      activatorContainer.current,
-    );
-    const focusableActivator: HTMLElement & {
-      disabled?: boolean;
-    } = firstFocusable || activatorContainer.current;
-
-    const activatorDisabled =
-      'disabled' in focusableActivator && Boolean(focusableActivator.disabled);
-
-    setActivatorAttributes(focusableActivator, {
-      id,
+const PopoverComponent = forwardRef<PopoverPublicAPI, PopoverProps>(
+  function Popover(
+    {
+      activatorWrapper = 'div',
+      children,
+      onClose,
+      activator,
+      preventFocusOnClose,
       active,
+      fixed,
       ariaHaspopup,
-      activatorDisabled,
-    });
-  }, [id, active, ariaHaspopup]);
-
-  const handleClose = (source: PopoverCloseSource) => {
-    onClose(source);
-    if (activatorContainer.current == null || preventFocusOnClose) {
-      return;
+      preferInputActivator = true,
+      colorScheme,
+      zIndexOverride,
+      ...rest
+    },
+    ref,
+  ) {
+    const [activatorNode, setActivatorNode] = useState<HTMLElement>();
+
+    const overlayRef = useRef<PopoverOverlay>(null);
+    const activatorContainer = useRef<HTMLElement>(null);
+
+    const WrapperComponent: any = activatorWrapper;
+    const id = useUniqueId('popover');
+
+    function forceUpdatePosition() {
+      overlayRef.current?.forceUpdatePosition();
     }
 
-    if (
-      (source === PopoverCloseSource.FocusOut ||
-        source === PopoverCloseSource.EscapeKeypress) &&
-      activatorNode
-    ) {
-      const focusableActivator =
-        findFirstFocusableNodeIncludingDisabled(activatorNode) ||
-        findFirstFocusableNodeIncludingDisabled(activatorContainer.current) ||
-        activatorContainer.current;
-      if (!focusNextFocusableNode(focusableActivator, isInPortal)) {
-        focusableActivator.focus();
+    useImperativeHandle(ref, () => {
```

Comment:
For anyone wondering - I've barely changes this file, but since we now wrap with a `forwardRef`, the indentation/formatting got updated and therefor its hard to really parse exact lines of code that have changed 😭